tensorboard/webapp/metrics/views/card_renderer/scalar_card_types.ts

```diff
@@ -116,18 +116,25 @@ export enum SortingOrder {
 }
 
 export interface SortingInfo {
-  header: ColumnHeaderType;
+  name: string;
   order: SortingOrder;
 }
 
+export interface TableDataPoint {
+  name: string;
+  value: string | number;
+}
+
 /**
  * An object which essentially contains the data for an entire row in the
- * DataTable. It will have a value for each required ColumnHeader for a given
- * run.
+ * DataTable.
+ * id: the run id for the run represented in this row
+ * data: the data for each column in the row
  */
 export type SelectedStepRunData = {
-  [key in ColumnHeaderType]?: string | number;
-} & {id: string};
+  id: string;
+  data: TableDataPoint[];
+};
```

Comment:
I think this list makes a lot of your other code a lot more complicated. Could this just be a record?

```suggestion
  export type SelectedStepRunData = {id: string} & Record<string, string|number>;
```